data/crac-io/crac-io-cne/src/main/java/com/farao_community/farao/data/crac_io_cne/Cne.java

```diff
@@ -0,0 +1,117 @@
+/*
+ *  Copyright (c) 2020, RTE (http://www.rte-france.com)
+ *  This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_io_cne;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_result_extensions.CracResultExtension;
+import com.farao_community.farao.data.crac_result_extensions.ResultVariantManager;
+import com.powsybl.iidm.network.Network;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.datatype.DatatypeConfigurationException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static com.farao_community.farao.data.crac_io_cne.CneClassCreator.*;
+import static com.farao_community.farao.data.crac_io_cne.CneConstants.*;
+import static com.farao_community.farao.data.crac_io_cne.CneUtil.*;
+
+/**
+ * Fills the classes that constitute the CNE file structure
+ *
+ * @author Viktor Terrier {@literal <viktor.terrier at rte-france.com>}
+ */
+public class Cne {
+    private CriticalNetworkElementMarketDocument marketDocument;
+    private CneHelper cneHelper;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Cne.class);
+
+    public Cne(Crac crac, Network network) {
+        marketDocument = new CriticalNetworkElementMarketDocument();
+        cneHelper = new CneHelper(crac, network);
+    }
+
+    public CriticalNetworkElementMarketDocument getMarketDocument() {
+        return marketDocument;
+    }
+
+    /*****************
+     GENERAL METHODS
+     *****************/
+    // Main method
+    public void generate() {
+
+        cneHelper.checkSynchronize();
+        Crac crac = cneHelper.getCrac();
+
+        fillHeader(crac.getNetworkDate());
+        addTimeSeriesToCne(crac.getNetworkDate());
+        Point point = marketDocument.getTimeSeries().get(0).getPeriod().get(0).getPoint().get(0);
+
+        if (crac.getExtension(CracResultExtension.class) != null && crac.getExtension(ResultVariantManager.class).getVariants() != null) { // Computation ended
```

Comment:
I am not sure this check is useful, and if it is, I think it should be done in the constructor of the CneHelper as well.